engine/collection/ingest/engine_test.go

```diff
@@ -181,12 +181,43 @@ func (suite *Suite) TestInvalidTransaction() {
 			invalidSig := unittest.InvalidFormatSignature()
 			tx := unittest.TransactionBodyFixture()
 			tx.ReferenceBlockID = suite.root.ID()
-			tx.PayloadSignatures[0] = invalidSig
+			tx.PayloadSignatures = []flow.TransactionSignature{invalidSig}
 
 			err := suite.engine.ProcessLocal(&tx)
 			suite.Assert().Error(err)
 			suite.Assert().True(errors.As(err, &access.InvalidSignatureError{}))
 		})
+
+		suite.Run("duplicated signature (envelope only)", func() {
+			tx := unittest.TransactionBodyFixture()
+			tx.ReferenceBlockID = suite.root.ID()
+			tx.EnvelopeSignatures = []flow.TransactionSignature{unittest.TransactionSignatureFixture(), unittest.TransactionSignatureFixture()}
```

Comment:
These are relying on somewhat non-obvious behaviour (that signature fixtures happen to always have the same address / key index). Might be worth a comment.

Reply:
💯 , I'll change it from fixture to explicitly define signatures here.

Reply:
updated at ccc1df0